crypto/bls_core.c

```diff
@@ -131,14 +131,171 @@ int bls_verify(const ep2_t pk, const byte* sig, const byte* data, const int len)
 
     int res = fp12_cmp(pair1, pair2);
 #endif
-    fp12_free(&one);
     ep_free(elemsG1[0]);
     ep_free(elemsG1[1]);
     ep2_free(elemsG2[0]);
     ep2_free(elemsG2[1]);
     
-    if (res == RLC_EQ && core_get()->code == RLC_OK) 
+    if (res == RLC_EQ && core_get()->code == RLC_OK) {
         return VALID;
-    else 
+    }
+    return INVALID;
+}
+
+
+// Verifies the validity of an aggregated BLS signature under distinct messages.
+//
+// Each message is mapped to a set of public keys, so that the verification equation is 
+// optimized to compute one pairing per message. 
+// - nb_hashes is the number of the messages (hashes) in the map
```

Comment:
>messages (hashes)
I think it corresponds to `sig`? right? they may we add this to the comment.

Reply:
`nb_hashes` is the number of distinct messages, there might be many.
There is only one signature `sig`. Did I understand your question correctly?